### PR TITLE
feat(CO-24): inventory page

### PR DESCRIPTION
**Pull request by dyanimay**
-created inventory-page AND inventory child-component
-added mock equip-button and item-detail component (fully commented because that is another task/branch)
PLEASE DO NOT COMMENT ON BUTTON AND ITEM-DETAIL COMPONENTS
-scss is without responsive view because that is another branch

### CODE REVIEW

File: src/app/components/inventory-page/inventory-page.component.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  private _id: number;
+  public _heroInventory: Equipment[];
+  selectedItem: Equipment;
+
+  constructor(private _getSingleHero: HeroService, private _route: ActivatedRoute ) {
+  this._id = Number(this._route.snapshot.paramMap.get('id'));
+  this._getSingleHero.getHero(this._id)
+    .subscribe(data => {
+      this._hero = data;
+      console.log(data.inventory);
```
Comment (Galicz555) [inline src/app/components/inventory-page/inventory-page.component.ts]: unnecessary console.log()

File: src/app/components/inventory-page/inventory-page.component.scss [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+.characterName {
+  text-align: center;
+}
+
+h4 {
+  text-transform: uppercase;
+  margin: 2% 0;
+}
```
Comment (Galicz555) [inline src/app/components/inventory-page/inventory-page.component.scss]: I don't know why but there is a scroll bar even though the number of items do not make it necessary. If it is possible make some changes :D

Comment (dyanimay) [reply src/app/components/inventory-page/inventory-page.component.scss]: thank you! solved with "overflow-y: auto;" in inventory.scss

File: src/app/components/inventory-page/inventory-page.component.ts (hunk 1 above)
Comment (dyanimay) [reply src/app/components/inventory-page/inventory-page.component.ts]: corrected

File: src/app/components/inventory-page/inventory-page.component.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  private _id: number;
+  public _heroInventory: Equipment[];
+  selectedItem: Equipment;
+
+  constructor(private _getSingleHero: HeroService, private _route: ActivatedRoute ) {
+  this._id = Number(this._route.snapshot.paramMap.get('id'));
+  this._getSingleHero.getHero(this._id)
+    .subscribe(data => {
+      this._hero = data;
+      console.log(this._hero);
```
Comment (juhaszPeterDev) [inline src/app/components/inventory-page/inventory-page.component.ts]: unnecessary comment

File: src/app/components/inventory-page/inventory-page.component.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+@Component({
+  selector: 'app-inventory-page',
+  templateUrl: './inventory-page.component.html',
+  styleUrls: ['./inventory-page.component.scss'],
+})
+export class InventoryPageComponent implements OnInit {
+  private _hero: HeroModel;
+  private _id: number;
+  public _heroInventory: Equipment[];
```
Comment (juhaszPeterDev) [inline src/app/components/inventory-page/inventory-page.component.ts]: this variable is redundancy, you can just as well get it from the hero field

File: src/app/components/inventory/inventory.component.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  valueChanged(equipment: Equipment) {
+    this.selectedItem = equipment;
+    this.valueChange.emit(this.selectedItem);
+  }
+
+  constructor() {}
+
+  ngOnInit() {}
+
+  handleclick(equipment: Equipment) {
```
Comment (juhaszPeterDev) [inline src/app/components/inventory/inventory.component.ts]: this seems to be an unused function, if so, delete it

File: src/app/components/item-detail/item-detail.component.html [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<!-- <div class="tab-content">
```
Comment (juhaszPeterDev) [inline src/app/components/item-detail/item-detail.component.html]: no

File: src/app/components/item-detail/item-detail.component.scss [hunk 7]
```diff
@@ -0,0 +1,8 @@
+// h4, button {
```
Comment (juhaszPeterDev) [inline src/app/components/item-detail/item-detail.component.scss]: no

File: src/app/components/item-detail/item-detail.component.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
+// import { Component, Input, OnInit } from '@angular/core';
```
Comment (juhaszPeterDev) [inline src/app/components/item-detail/item-detail.component.ts]: no

File: src/app/components/register/register.component.spec.ts [hunk 9]
```diff
@@ -14,11 +14,12 @@ import { NotFoundComponent } from '../not-found/not-found.component';
 import { LoginPageComponent } from '../login-page/login-page.component';
```
Comment (juhaszPeterDev) [inline src/app/components/register/register.component.spec.ts]: refactor this spec file and neuter it

File: src/app/components/inventory-page/inventory-page.component.ts (hunk 3 above)
Comment (dyanimay) [reply src/app/components/inventory-page/inventory-page.component.ts]: deleted

File: src/app/components/inventory/inventory.component.ts (hunk 5 above)
Comment (dyanimay) [reply src/app/components/inventory/inventory.component.ts]: deleted

File: src/app/components/inventory-page/inventory-page.component.ts (hunk 4 above)
Comment (dyanimay) [reply src/app/components/inventory-page/inventory-page.component.ts]: true, deleted and changed

File: src/app/components/register/register.component.spec.ts (hunk 9 above)
Comment (dyanimay) [reply src/app/components/register/register.component.spec.ts]: refactored

**Pull request merged**